CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -1310,20 +1310,30 @@ public function assignFormVariablesByContributionID() {
       return;
     }
     if (!$this->getContactID()) {
-      CRM_Core_Error::statusBounce(ts("Returning since there is no contact attached to this contribution id."));
+      throw new CRM_Core_Exception(ts("Returning since there is no contact attached to this contribution id."));
     }
 
     $payment = CRM_Contribute_BAO_Contribution::getPaymentInfo($this->_ccid, 'contribution');
     //bounce if the contribution is not pending.
-    if (empty($payment['balance'])) {
-      CRM_Core_Error::statusBounce(ts("Returning since contribution has already been handled."));
+    if (empty($payment['balance']) || !in_array($payment['contribution_status'], array('Pending', 'Partially paid'))) {
+      throw new CRM_Core_Exception(ts("This contribution is already fully paid."));
     }
-    if (!empty($payment['total'])) {
-      $this->_pendingAmount = $payment['total'];
+
+    $paymentInfo = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType($this->_ccid, 'contribution');
+    $pendingAmount = CRM_Utils_Array::value('amount_owed', $paymentInfo);
```

Comment:
I looked specifically at this part and I think we can make a preliminary change to support this - see 

https://github.com/civicrm/civicrm-core/pull/13151

which means we will be supporting the 'correct' amount at this point.

Reply:
Cool - as @monishdeb is already reviewing that - will update the call here as soon as it is merged.